tests/js/common/replication_api/shell-api-replication.js

```diff
@@ -0,0 +1,2524 @@
+      assertEqual(all["database"], "bar");
+      assertEqual(all["requestTimeout"], 12.5);
+      assertEqual(all["connectTimeout"], 26.3);
+      assertEqual(all["ignoreErrors"], 2);
+      assertEqual(all["maxConnectRetries"], 12);
+      assertEqual(all["sslProtocol"], 0);
+      assertEqual(all["chunkSize"], 1234567);
+      assertFalse(all["autoStart"]);
+      assertTrue(all["adaptivePolling"]);
+      assertFalse(all["autoResync"]);
+      assertFalse(all["includeSystem"]);
+      assertFalse(all["requireFromPresent"]);
+      assertFalse(all["verbose"]);
+      assertEqual(all["connectionRetryWaitTime"], 2.5);
+      assertEqual(all["initialSyncMaxWaitTime"], 4.3);
+      assertEqual(all["idleMinWaitTime"], 0.22);
+      assertEqual(all["idleMaxWaitTime"], 3.5);
+    },
+
+    ////////////////////////////////////////////////////////////////////////////////;
+    // state;
+    ////////////////////////////////////////////////////////////////////////////////;
+
+    test_checks_the_applier_state_again_2: function() {
+      // fetch state;
+      let cmd = api + "/applier-state";
+      let doc = arango.GET_RAW(cmd);
+
+      assertEqual(doc.code, 200);
+      let all = doc.parsedBody;
+      assertTrue(all.hasOwnProperty('state'));
+      assertTrue(all.hasOwnProperty('server'));
+
+      let state = all['state'];
+      assertFalse(state['running']);
+      assertTrue(state.hasOwnProperty("lastAppliedContinuousTick"));
+      assertTrue(state.hasOwnProperty("lastProcessedContinuousTick"));
+      assertTrue(state.hasOwnProperty("lastAvailableContinuousTick"));
+      assertTrue(state.hasOwnProperty("safeResumeTick"));
+
+      assertTrue(state.hasOwnProperty("progress"));
+      let progress = state['progress'];
+      assertTrue(progress.hasOwnProperty("time"));
+      assertMatch(/^([0-9]+-[0-9]+-[0-9]+T[0-9]+:[0-9]+:[0-9]+Z)?$/, progress['time']);
+      assertTrue(progress.hasOwnProperty("failedConnects"));
+
+      assertTrue(state.hasOwnProperty("totalRequests"));
+      assertTrue(state.hasOwnProperty("totalFailedConnects"));
+      assertTrue(state.hasOwnProperty("totalEvents"));
+      assertTrue(state.hasOwnProperty("totalOperationsExcluded"));
+
+      assertTrue(state.hasOwnProperty("lastError"));
+      let lastError = state["lastError"];
+      assertTrue(lastError.hasOwnProperty("errorNum"));
+      assertTrue(state.hasOwnProperty("time"));
+      assertMatch(/^[0-9]+-[0-9]+-[0-9]+T[0-9]+:[0-9]+:[0-9]+Z$/, state['time']);
+    },
+  };
+}
+
+////////////////////////////////////////////////////////////////////////////////;
+// logger;
+////////////////////////////////////////////////////////////////////////////////;
+function dealing_with_the_logger_Suite () {
+  return {
+    setUp: function() {
+      db._drop("UnitTestsReplication");
+      let res = db._createDatabase("UnitTestDB");;
+      assertTrue(res);
+      arango.PUT_RAW(api + "/applier-stop", "");
+      arango.DELETE_RAW(api + "/applier-state", "");
```

Comment:
I don't know if this works correctly.
The testsuite creates a database, but does not use it here. So calling the functions on the applier will be carried out in the system database.

Reply:
Additionally, what is the value of `api` here? The test suite doesn't set it, so this will use the global `api` variable, not the test suite-specific one. This is highly confusing to me.